server/src/metadata/field-metadata/field-metadata.service.ts

```diff
@@ -123,6 +139,66 @@ export class FieldMetadataService extends TypeOrmQueryService<FieldMetadataEntit
     return createdFieldMetadata;
   }
 
+  override async updateOne(
+    id: string,
+    record: UpdateFieldInput,
+  ): Promise<FieldMetadataEntity> {
+    const existingFieldMetadata = await this.fieldMetadataRepository.findOne({
+      where: {
+        id,
+        workspaceId: record.workspaceId,
+      },
+    });
+
+    if (!existingFieldMetadata) {
+      throw new NotFoundException('Field does not exist');
+    }
+
+    const objectMetadata =
+      await this.objectMetadataService.findOneWithinWorkspace(
+        existingFieldMetadata?.objectMetadataId,
+        record.workspaceId,
+      );
+
+    if (!objectMetadata) {
+      throw new NotFoundException('Object does not exist');
+    }
+
+    // Check if the id of the options has been provided
+    if (record.options) {
+      for (const option of record.options) {
+        if (!option.id) {
+          throw new BadRequestException('Option id is required');
+        }
+      }
+    }
+
+    const updatedFieldMetadata = await super.updateOne(id, record);
+
+    if (record.options || record.defaultValue) {
+      await this.workspaceMigrationService.createCustomMigration(
```

Comment:
I am not sure about the createCustomMigration API long term. I feel that we should maybe provide something more restrictive: createColumn, udpateColumnType, updateEnumOptions...